vllm/entrypoints/openai/api_server.py

```diff
@@ -70,50 +62,17 @@ async def check_model(request) -> Optional[JSONResponse]:
 
 
 async def get_gen_prompt(request) -> str:
-    if not _fastchat_available:
-        raise ModuleNotFoundError(
-            "fastchat is not installed. Please install fastchat to use "
-            "the chat completion and conversation APIs: `$ pip install fschat`"
-        )
-    if version.parse(fastchat.__version__) < version.parse("0.2.23"):
-        raise ImportError(
-            f"fastchat version is low. Current version: {fastchat.__version__} "
-            "Please upgrade fastchat to use: `$ pip install -U fschat`")
-
-    conv = get_conversation_template(request.model)
-    conv = Conversation(
-        name=conv.name,
-        system_template=conv.system_template,
-        system_message=conv.system_message,
-        roles=conv.roles,
-        messages=list(conv.messages),  # prevent in-place modification
-        offset=conv.offset,
-        sep_style=SeparatorStyle(conv.sep_style),
-        sep=conv.sep,
-        sep2=conv.sep2,
-        stop_str=conv.stop_str,
-        stop_token_ids=conv.stop_token_ids,
-    )
-
-    if isinstance(request.messages, str):
-        prompt = request.messages
+    if chat_template is not None:
+        return tokenizer.apply_chat_template(conversation=request.messages,
+                                             chat_template=chat_template,
+                                             tokenize=False)
```

Comment:
@Tostino  should we add add_generation_prompt=True for generation prompt?

Reply:
Uh, yeah I am pretty sure we should. Let me do a little research tomorrow on templates that properly support it. The template needs to actually use that variable somewhere in it for it to take any effect. I had a long travel day today, will get back to you tomorrow some time.

Reply:
afaik `add_generation_prompt` is currently only needed for mistral (or at least I tested with mistral). For llama you don't need add_generation_prompt

Probably should allow user to control this, maybe through another cli flag `--add-generation-prompt/--no-add-generation-prompt` (default to False)?

What do you think?

Reply:
Okay, I believe I have resolved this now with the latest commit. This is now part of the request. I agree that we should allow users to set default values for those parameters through CLI flags, but think we should do that in it's own PR for all of the parameters that make sense.